src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -246,10 +262,63 @@ void SensorFusion::updateWorld(const SSLProto::SSL_DetectionFrame &ssl_detection
         enemy_team    = createEnemyTeam(yellow_team);
     }
 
-    new_ball = createBall(ball_detections);
-    if (new_ball)
+    ball_in_dribbler_timeout--;
+    if (ball_in_dribbler_timeout <= 0)
+    {
+        friendly_robot_id_with_ball_in_dribbler = std::nullopt;
+        ball_in_dribbler_timeout                = 0;
+    }
+
+    if (friendly_robot_id_with_ball_in_dribbler.has_value())
     {
-        updateBall(*new_ball);
+        std::optional<Robot> robot_with_ball_in_dribbler =
+            friendly_team.getRobotById(friendly_robot_id_with_ball_in_dribbler.value());
+
+        if (robot_with_ball_in_dribbler.has_value())
+        {
+            std::vector<BallDetection> dribbler_in_ball_detection = {BallDetection{
+                .position =
+                    robot_with_ball_in_dribbler->position() +
+                    Vector::createFromAngle(robot_with_ball_in_dribbler->orientation())
+                        // MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT of the ball should be
+                        // inside the robot
+                        .normalize(DIST_TO_FRONT_OF_ROBOT_METERS -
+                                   BALL_MAX_RADIUS_METERS *
+                                       MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT),
```

Comment:
Looking to `MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT`, I feel like the calculation for ball position should be something more like:
```
ball_center = DIST_TO_FRONT_OF_ROBOT_METERS + BALL_MAX_RADIUS_METERS - 2 * BALL_MAX_RADIUS_METERS * MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT
```
Mainly because `MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT` seems to be relative to the diameter instead of the radius of the ball.

Reply:
fixed